website/content/en/docs/main/tasks/private-networks/index.md

```diff
@@ -0,0 +1,394 @@
+
+To create connectivity between the ClusterLink gateways, we need to set up one FRP server with a public IP and create an FRP client for each ClusterLink gateway, as illustrated below.
+
+This task includes instructions on how to connect the peers using FRP. Instructions for creating full connectivity between applications to remote services can be found in the [Nginx tutorial][] and [iPerf3 tutorial][].
+
+In this task, we will extend the peer connectivity instructions to use FRP.
+
+## Create FRP Server
+
+In this step, we will create the FRP server on the same cluster we use for ClusterLink, but it can be on any peer or Kubernetes cluster.
+
+1. Create a configmap that contains the server configuration:
+
+    ```sh
+    echo "
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+        name: frps-config
+        namespace: clusterlink-system
+    data:
+        frps.toml: |
+            bindPort = 4443
+    " | kubectl apply -f -
+    ```
+
+    In this setup, we expose the FRP server pod on port `4443`.
+2. Create FRP server deployment:
+
+    ```sh
+    echo "
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: frps
+      namespace: clusterlink-system
+    spec:
+      replicas: 1
+      selector:
+        matchLabels:
+          app: frps
+      template:
+        metadata:
+          labels:
+            app: frps
+        spec:
+          hostNetwork: true
+          containers:
+            - name: frps
+              image: snowdreamtech/frps
+              volumeMounts:
+                - name: frps-config-volume
+                  mountPath: /etc/frp/frps.toml
+                  subPath: frps.toml
+          volumes:
+            - name: frps-config-volume
+              configMap:
+                name: frps-config
+          restartPolicy: Always
+    " | kubectl apply -f -
+    ```
+
+3. Create sn ingress service to expose the FRP server:
+
+    ```sh
+    echo "
+    apiVersion: v1
+    kind: Service
+    metadata:
+        name: clusterlink-frps
+        namespace: clusterlink-system
+    spec:
+        type: NodePort
+        selector:
+            app: frps
+        ports:
+          - port: 4443
+            targetPort: 4443
+            nodePort: 30444
+    " | kubectl apply -f -
+    ```
+
+    In this case, we use a `NodePort` service, but it can be other types like `LoadBalancer`.
+
+## Create FRPs Clients
+
+1. Set the `FRP_SERVER_IP` variable for each cluster:
+
+    *Client cluster*:
+
+    ```sh
+    export FRP_SERVER_IP=`docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' client-control-plane`
+    ```
+
+    *Client cluster*:
```

Comment:
why is this `*Client cluster*` duplicated? Can't be done once? if multiple terminals, then please be explicit